src/browser.ts

```diff
@@ -62,9 +62,14 @@ export interface AnalyticsBrowserSettings extends AnalyticsSettings {
   cdnSettings?: LegacySettings & Record<string, unknown>
 }
 
-export function loadLegacySettings(writeKey: string): Promise<LegacySettings> {
-  const cdn = window.analytics?._cdn ?? getCDN()
-  return fetch(`${cdn}/v1/projects/${writeKey}/settings`)
+export function loadLegacySettings(
+  writeKey: string,
+  cdn?: string
+): Promise<LegacySettings> {
+  if (cdn) setGlobalCDNUrl(cdn) // this is an ugly side-effect, but it's for the benefits of the plugins that get their cdn via getCDN()
```

Comment:
What do you think about moving the `setGlobalCDNUrl` to the `AnalyticsBrowser.load` method? I think the side-effect here of setting the global CDN on `analytics` makes more sense from the class that's actually creating that `analytics` object.

This function could be called multiple times with different CDN values (not something I'd ever expect to actually hapen), so removing the side-effect here makes this feel safer.

Reply:
Sure!